server/parameters/util/YAMLParser.java

```diff
@@ -258,6 +265,40 @@ Restricted<T> asRestricted() {
             }
         }
 
+        public static class Optional<T> extends Value<java.util.Optional<T>> {
```

Comment:
New optional value parser that wraps a primitive value parser, the say way `Restricted` does.